scipy/special/_support_cupy_torch_jax.py

```diff
@@ -0,0 +1,50 @@
+import sys
+from scipy._lib.array_api_compat.array_api_compat import array_namespace
+from . import _ufuncs
+from ._ufuncs import *
+
+__all__ = _ufuncs.__all__
+
+array_api_comat_prefix = "scipy._lib.array_api_compat.array_api_compat"
+
+
+def get_array_special_func(f_name, xp):
+    if xp.__name__ == f"{array_api_comat_prefix}.numpy":
+        f = getattr(_ufuncs, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.torch":
+        f = getattr(xp.special, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.cupy":
+        import cupyx
+        f = getattr(cupyx.scipy.special, f_name, None)
+    elif xp.__name__ == f"{array_api_comat_prefix}.jax":
+        f = getattr(xp.scipy.special, f_name, None)
+    else:
+        f = None
+
+    if f is None:
+        raise ValueError("SciPy cannot dispatch to special function "
+                         f"'{f_name}' in array library '{xp.__name__}'.")
```

Comment:
It could be nice to offer different messages for functions which are missing but one of (a) required by the spec, (b) optional in the spec, (c) defined in a later version of the spec than implemented by this library, or (d) not part of the spec.  It's annoying to implement but nice for users, at least...

Reply:
There is no spec for special functions right now. When it is written, we can consider making a distinction.

Reply:
Ah, I'd just append "because it is not part of the array API standard" to this message then (ie (d))

Reply:
Ok, I will be more specific. (There is no standard, and the library underlying the array does not happen to define the function.)

Reply:
It sounds like this error is to be replaced with conversion to NumPy array, evaluation of the function, and conversion back.